File: examples/objects/C#/Book.cs
using System;
namespace bookExample {

	/// <summary>
	/// A Object holding the price, author, and title of a book
	/// </summary>
	/// <remarks>
	/// Modified example from https://thimble.mozilla.org/en-US/anonymous/1dbb9564-0bc8-4a2e-94e8-ba6f6690fb69/304196
	/// </remarks>
	public class Book {

	  /// <summary>
	  /// Stores the author property as a String
	  /// </summary>
	  private String author;

	  /// <summary>
	  /// Stores the title property as a String
	  /// </summary>
	  private String title;

	  /// <summary>
	  /// Stores the price property as a float
	  /// </summary>
	  private double price;

	  /// <summary>
	  /// The class constructor
	  /// </summary>
	  /// <param name="author"> Initial author of the book</param>
	  /// <param name="title"> Initial title of the book</param>
	  public Book (string author, string title){
		this.price = 5.00;
		this.author = author;
		this.title = title;
	  }

	  /// <summary>
	  /// The class constructor
	  /// </summary>
	  /// <param name="author"> Initial author of the book</param>
	  /// <param name="title"> Initial title of the book</param>
	  /// <param name="price"> Initial price of the book</param>
	  public Book (string author, string title, double price){
		this.price = price;
		this.author = author;
		this.title = title;
	  }

	  /// <summary>
	  /// Returns the author of the book
	  /// </summary>
	  /// <returns> The author of the book as a string </returns>
	  public string getAuthor() {
		return this.author;
	  }

	  /// <summary>
	  /// Returns the title of the book
	  /// </summary>
	  /// <returns> The title of the book as a string </returns>
	  public string getTitle() {
		return this.title;
	  }

	  /// <summary>
	  /// Returns the price of the book
	  /// </summary>
	  /// <returns> The price of the book as a double </returns>
	  public double getPrice() {
		return price;
	  }

	  /// <summary>
	  /// Attempts to increase the price of the book
	  /// </summary>
	  /// <param="increase"> Price to increase by </param>
	  /// <returns> true if it is successful in modifying the price, false otherwise. </returns>
	  public bool increasePrice(double increase) {
		if (increase < 0) { // this is a decrease, not an increase
		  return false;
		}

		price = price + increase;
		return true;
	  }
	}
}

File: examples/objects/C++/book.cpp
/**
 * A Object holding the price, author, and title of a book
 * 
 * @author Mr. Seidel
 * @since 23-Sep-2018
 * @version 1.0
 * 
 * Updated 13-Apr-20:
 *  - "printX" methods
 *  + "getX" methods
 *  + updated main() function
 */
class Book {
    private:
        string author, title;  
        double price;		

    public:
	    Book(string, string);
	    Book(string, string, double);
        ~Book();
	    string getAuthor(void);
	    string getTitle(void);
	    double getPrice(void);
	    bool increasePrice(double);
};  // <--- note the semi-colon here

	
/**
* This is the constructor function for new books
* 
* @param author - This is the initial author
* @param title - This is the initial title of the book
*
*/
Book::Book (string _author, string _title){
	price = 0.00;
	author = _author;
	title = _title;
}

/**
* This is the constructor function for new books
* 
* @param _author - This is the initial author
* @param _title - This is the initial title of the book
* @param _price - This is the initial price of the book
*
*/
Book::Book (string _author, string _title, double _price){
	price = _price;
	author = _author;
	title = _title;
}

/**
* Returns the author of the book
*
* @return the author of the book
*/
string Book::getAuthor() {
	return author;
}

/**
* Returns the title of the book
* 
* @return the title of the book
*/
string Book::getTitle() {
	return title;
}

/**
* Returns the price of the book as a double
*
* @return the price of the book
*/
double Book::getPrice() {
	return price;
}

/**
* Attempts to increase the price of the book
* 
* @param increase This is the amount to increase the price of the book
* @return true if it is successful in modifying the price, false otherwise.
*/
bool Book::increasePrice(double increase) {
	if (increase < 0) {	// this is a decrease, not an increase
		return false;
	}
	price = price + increase;
	return true;
}